lib/ansible/module_utils/urls.py

```diff
@@ -1479,7 +1520,26 @@ def open(self, method, url, data=None, headers=None, use_proxy=None,
             else:
                 request.add_header(header, headers[header])
 
-        return urllib_request.urlopen(request, None, timeout)
+        r = urllib_request.urlopen(request, None, timeout)
+        if decompress and r.headers.get('content-encoding', '').lower() == 'gzip':
+            fp = GzipDecodedReader(r.fp)
```

Comment:
right now this will error out, wouldn't a better option be to save the compressed content as a compressed file?

Reply:
I'm on the fence about this. I think the most useful option is to decompress by default, but it is a change in behavior potentially, although limited in impact.  The original reporters have classified this as a bug.

I suppose I could leave the as default enabled, but if `HAS_GZIP` is `False` flip `decompress` to `False` and warn.

Otherwise, I think we'd need to flip the default of `decompress` to `False`, which I think is less useful to users.

Reply:
Ok, I think I have this and the above comment resolved.